pkg/api/validation/validation.go

```diff
@@ -1538,6 +1548,15 @@ func ValidatePersistentVolumeClaimSpec(spec *api.PersistentVolumeClaimSpec, fldP
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("storageClassName"), *spec.StorageClassName, msg))
 		}
 	}
+	if spec.VolumeMode != nil && *spec.VolumeMode == api.PersistentVolumeBlock && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("volumeMode"), "PersistentVolumeClaim volumeMode is disabled by feature-gate"))
+		return allErrs
+	}
+	if spec.VolumeMode != nil && len(*spec.VolumeMode) > 0 {
```

Comment:
This implies that "" is valid value. I think that only "Filesystem" and "Block" should be allowed.

Reply:
@jsafrane  - are you saying you want me to combine the sub if line 1555 and 1556?

Reply:
Remove the len() check